src/NHibernate/Cache/ICacheConcurrencyStrategy.cs

```diff
@@ -134,13 +135,17 @@ public partial interface ICacheConcurrencyStrategy
 		/// </summary>
 		string RegionName { get; }
 
+		// 6.0 TODO: type as BaseCache instead
+#pragma warning disable 618
```

Comment:
Implementors will be warned by the `ICache` obsolete message about that, but they will not be able to actually prepare for it. I do not see a way to avoid this. The `HashtableCacheProvider.BuildCache` implementation pattern commented previously is not usable here, as we may have a cache not deriving from `CacheBase` here.